### ISSUE TITLE
`selectedRange` is inconsistent with CRLF

### ISSUE THREAD

**Issue by umireon** [coteditor/CotEditor#659]
## Environment

- CotEditor: 3.1.3-alpha (178)
- System: macOS Version 10.12.1 (Build 16B2657)
- Language: English

## Short Description
In `get` of `Editable.selectedRange`,  line endings are counted as two character (CRLF).
In `set` of that, line endings are counted as one character (LF).
This inconsistency causes odd behavior for scripts which handles selection.

## Steps to Reproduce the Issue
1. Open or create a document with `CRLF`.
2. Execute the following script:

```applescript
tell front document of application "CotEditor"
    -- Essentially noop script
    set theRange to range of selection
    set range of selection to theRange
end tell
```

3. The selected range moves contrary to expectation.

## Expected Result
Nothing happens.

## Related codes

https://github.com/coteditor/CotEditor/blob/14d74d1bfe375b153ccff7835811717cc94c3d7c/CotEditor/Sources/Editable.swift#L56-L76

`self.textView.string` is assumed to have the actual line endings but always have `LF`s.

**Comment by umireon** [comment]
I didn't notice there is an issue template.

**Comment by 1024jp** [comment]
> `self.textView.string` is assumed to have the actual line endings but always have LFs.

`self.textView.string`, equal `EditorTextView`, is designed to have always LF. I sanitize all line endings before passing strings to textEditor and vice versa.
// Actually, I don't know why it is so designed. I'm just keeping the original design since I adopted CotEditor project.
The Scripting interface of `document` object pretends that the content string has the actual line endings.

So, Editable protocol is correct. However, as you pointed out, something is wrong with selected range on scripting. There should be an issue with the range conversion but somewhere else. Probably in TextSelection or around there. I'll look into it later (Due to my personal reason, I'm quite busy in this week. Next week, I'll be able to have more development time).

**Comment by umireon** [comment]
Thank you for following up @1024jp !
FYI i have created a possible fix as a PR (#660).
